src/core/ext/xds/xds_api.cc

```diff
@@ -1685,19 +1685,25 @@ grpc_error* LdsResponseParseClient(
             absl::StrCat("duplicate HTTP filter name: ", name).c_str());
       }
       names_seen.insert(name);
+      const bool is_optional =
+          envoy_extensions_filters_network_http_connection_manager_v3_HttpFilter_is_optional(
+              http_filter);
       const google_protobuf_Any* any =
           envoy_extensions_filters_network_http_connection_manager_v3_HttpFilter_typed_config(
               http_filter);
+      if (any == nullptr) {
+        if (is_optional) continue;
+        return GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+            absl::StrCat("no filter config specified for filter name ", name)
+                .c_str());
+      }
```

Comment:
We should update `ParseTypedPerFilterConfig` with the same logic as well.

Reply:
Thanks for pointing this out.  That's not actually possible to be null, because it's a map, so upb will return an empty object as the value even if the proto doesn't have the field populated.  But we probably should return a better error message in that case.

Fixed.